cpp/dolfinx/mesh/generation.h

```diff
@@ -93,6 +100,15 @@ Mesh<T> create_box(MPI_Comm comm, MPI_Comm subcomm,
                    std::array<std::int64_t, 3> n, CellType celltype,
                    CellPartitionFunction partitioner = nullptr)
 {
+  if (std::ranges::any_of(n, [](auto e) { return e < 1; }))
```

Comment:
Mainline clang and gcc are pretty good. It's Apple's clang which is always running behind, particularly with std::ranges. And we need to support Apple clang (conda).

My preference would be to try std::ranges on a small part of Basix first, then if it's successful port DOLFINx across in small chunks - I think it will hugely improve readability.

Reply:
I remove the usage of `std::ranges` for now from this PR?

Reply:
Opened a PR in `basix` adressing this, see https://github.com/FEniCS/basix/pull/837

Reply:
Should be resolved when we have an answer on https://github.com/FEniCS/dolfinx/pull/3293 - adding dependency.